### ISSUE TITLE
Inconsistent PSR-15 request manipulation

### ISSUE THREAD

**Issue by oojacoboo** [Ecodev/graphql-upload#7]
When using this lib in middleware, the request body is parsed and re-attached to the request object that's piped through, if there are uploadedFiles.  However, if there are not, the request is not parsed.  Additionally, the stream is no longer available in the request object that's piped through the middleware, so you're unable to rewind and reparse, if needed.  

This inconsistency is causing issues.  I get that the request object needs to be modified, but the stream should at least be maintained, at the least.  Or maybe just don't return a parsedBody at all.  The real issue is that the parsing of the body isn't deep in it's JSON decoding.  So, you can't even use it and you have to do a whole bunch of validation checks on the state of the parsedBody for it to be usable.

**Comment by PowerKiKi** [comment]
I am not quite sure to understand how you are using this middleware and were your issues arise. This lib is meant to be used with `webonyx/graphql-php`. Is it not what you are doing ? could you show with a [reprex](https://stackoverflow.com/help/minimal-reproducible-example) what you're doing and what does not work as intended ?

**Comment by oojacoboo** [comment]
@PowerKiKi It's being used with `webonyx/graphql-php`.  But, we also have other middleware.

The issue is that the request object doesn't have the stream anymore and the `parsedBody` includes this:

```php
$result = json_decode($bodyParams['operations'], true);
if (isset($result['operationName'])) {
    $result['operation'] = $result['operationName'];
    unset($result['operationName']);
}
```

Why is there an `operation` key added to the `parsedBody`?  Is this a GraphQL standard?  I don't see a request body in this format from typical clients.

**Comment by oojacoboo** [comment]
I was able to get something working in our case.  But, the addition of the `operation` key into the request body was troublesome:

```php
$result = json_decode($bodyParams['operations'], true);
if (isset($result['operationName'])) {
    $result['operation'] = $result['operationName'];
    unset($result['operationName']);
}
```

Is this a GraphQL standard?  Without this lib, the request body does not have this, at least with the clients I've tested.  Also, I was unable to request the request stream on the request object to "re-parse" the body as I saw fit in subsequent middleware.  Essentially, the initial request body wasn't available via the request object in the pipeline.
